apps/microsoft-teams/app-actions/src/actions/handle-app-event.ts

```diff
@@ -21,14 +25,63 @@ export const handler = withAsyncAppActionErrorHandling(
   async (
     parameters: AppActionCallParameters,
     context: AppActionCallContext
-  ): Promise<AppActionCallResponse<SendEntryActivityMessageResult[]>> => {
+  ): Promise<
+    AppActionCallResponse<SendEntryActivityMessageResult[] | SendWorkflowUpdateMessageResult>
+  > => {
     const {
       cma,
       appActionCallContext: { appInstallationId, environmentId, spaceId, userId, cmaHost },
     } = context;
 
     const { payload, topic: topicString, eventDatetime } = parameters;
 
+    /**
+     * In the interest of time, as well as prototyping/learning we made the explicit choice
+     * to re-use/extend this "app-events" app-action to handle "workflow" events as well.
+     *
+     * This block of code to handle workflow topics is intentionally repetative and intrusive.
+     * The end goal is to create a new app action to specifically handle "workflow" events.
+     * Ideally, at that time, it will be a simple copy/paste operation, leaving this
+     * app-action unchanged.
+     */
+    if (topicString === TOPIC_ACTION_MAP['Workflow.Step.notifyMicrosoftTeams']) {
```

Comment:
Hmm. Since this is really a "hack" I suggest we we just compare to the literal value here `Workflow.Step.notifyMicrosoftTeams` and not try to add this to the TOPIC_ACTION_MAP. It doesn't belong there.

Reply:
Hmmm... In my weakly held opinion, I'd still prefer to add `Workflow.Step.notifyMicrosoftTeams` to the list of topics that this app (not app action) can expect.  The downside of that is that a workflow topic, doesn't really _map_ to anything in the way that app events do.

All that being said, I'm getting a bunch of type errors, because `selectedEvents` and `helpers.buildEntryActivity` are expecting their topic param to be an `app-event` topic.  In the interest of time, I'm going to make the tradeoff to shoe-horn `Workflow.Step.notifyMicrosoftTeams` into the list of _until now_ app-event topics.  Hopefully this can be cleaned up when we address the above comment/todo.